example_test.go

```diff
@@ -1172,9 +1239,16 @@ func ExampleConnection_Do_failure() {
 	future := conn.Do(req)
 
 	// When the future receives the response, the result of the Future is set
-	// and becomes available. We could wait for that moment with Future.Get()
-	// or Future.GetTyped() methods.
-	resp, err := future.Get()
+	// and becomes available. We could wait for that moment with Future.Get(),
+	// Future.GetResponse() or Future.GetTyped() methods.
+	resp, err := future.GetResponse()
+	if err != nil {
+		fmt.Printf("Error in the future: %s\n", err)
+	}
+
+	data, err := future.Get()
+	fmt.Printf("Data: %v\n", data)
```

Comment:
Nit, please check the resp.Header().Error first. And write a comment that it is an optional step.